coremltools/optimize/torch/quantization/_annotation_config.py

```diff
@@ -0,0 +1,109 @@
+#  Copyright (c) 2024, Apple Inc. All rights reserved.
+#
+#  Use of this source code is governed by a BSD-3-clause license that can be
+#  found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+
+from typing import Optional as _Optional
+
+import torch as _torch
```

Comment:
Is it safe to assume that torch is installed here? People can have coremltools installed without having torch installed, ex: maybe you're using coremltools to convert TensorFlow models.

Reply:
These are under `coremltools.optimize.torch`, we are already doing this in existing code: https://github.com/apple/coremltools/blob/b416f36054af9ca9d10b2d74ba215d0454677ca0/coremltools/optimize/torch/quantization/quantization_config.py#L19. Since it's under `coremltools.optimize.torch`, it's only meant to be used with PyTorch. 

This is not imported at the module level (in any `__init__.py` file), so it shouldn't break users who are only using TensorFlow.